pkgs/electrs/default.nix

```diff
@@ -1,2933 +1,25 @@
-            usesDefaultFeatures = false;
-          };
-        };
-        features = {
-          "default" = [ "proc-macro" ];
-          "proc-macro" = [ "proc-macro2/proc-macro" ];
-        };
-        resolvedDefaultFeatures = [ "default" "proc-macro" "proc-macro2" ];
-      };
-    "rand 0.3.23 (registry+https://github.com/rust-lang/crates.io-index)"
-      = rec {
-        crateName = "rand";
-        version = "0.3.23";
-        edition = "2015";
-        sha256 = "118rairvv46npqqx7hmkf97kkimjrry9z31z4inxcv2vn0nj1s2g";
-        authors = [
-          "The Rust Project Developers"
-        ];
-        dependencies = {
-          "libc" = "libc 0.2.54 (registry+https://github.com/rust-lang/crates.io-index)";
-          "rand" = "rand 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)";
-        };
-        features = {
-          "nightly" = [ "i128_support" ];
-        };
-      };
-    "rand 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)"
-      = rec {
-        crateName = "rand";
-        version = "0.4.6";
-        edition = "2015";
-        sha256 = "0c3rmg5q7d6qdi7cbmg5py9alm70wd3xsg0mmcawrnl35qv37zfs";
-        authors = [
-          "The Rust Project Developers"
-        ];
-        dependencies = {
-          "fuchsia-cprng" = {
-            packageId = "fuchsia-cprng 0.1.1 (registry+https://github.com/rust-lang/crates.io-index)";
-            target = (target."os" == "fuchsia");
-          };
-          "libc" = {
-            packageId = "libc 0.2.54 (registry+https://github.com/rust-lang/crates.io-index)";
-            optional = true;
-            target = target."unix";
-          };
-          "rand_core" = {
-            packageId = "rand_core 0.3.1 (registry+https://github.com/rust-lang/crates.io-index)";
-            usesDefaultFeatures = false;
-            target = (target."env" == "sgx");
-          };
-          "rdrand" = {
-            packageId = "rdrand 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)";
-            target = (target."env" == "sgx");
-          };
-          "winapi" = {
-            packageId = "winapi 0.3.7 (registry+https://github.com/rust-lang/crates.io-index)";
-            target = target."windows";
-            features = [ "minwindef" "ntsecapi" "profileapi" "winnt" ];
-          };
-        };
-        features = {
-          "default" = [ "std" ];
-          "nightly" = [ "i128_support" ];
-          "std" = [ "libc" ];
-        };
-        resolvedDefaultFeatures = [ "default" "libc" "std" ];
-      };
-    "rand_core 0.3.1 (registry+https://github.com/rust-lang/crates.io-index)"
-      = rec {
-        crateName = "rand_core";
-        version = "0.3.1";
-        edition = "2015";
-        sha256 = "0q0ssgpj9x5a6fda83nhmfydy7a6c0wvxm0jhncsmjx8qp8gw91m";
-        authors = [
-          "The Rand Project Developers"
-          "The Rust Project Devel
```

Comment:
Do you know what exactly is the input to the hash? There seemed to have been problems with fully pinning the dependencies before we used crate2nix (https://github.com/fort-nix/nix-bitcoin/issues/66 @jb55). Other than that I'd be very happy to simplify this by going back to buildRustPackage.

Reply:
You mean `cargoSha256`? It's the fixed output hash on the result of running `cargo vendor`. (Here's the [derivation source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/fetchCargoTarball.nix).)
 With the current state of `buildRustPackage` the crate2nix approach has no advantages when it comes to pinning. Both fetch the exact dependencies defined in cargo.lock. 

To quote @jb55, the reason for switching to crate2nix was

> only to find that the rust packages in whatever rust channel nixpkgs was using was
> out of date

crate2nix doesn't pin `rustPackages` (that is rustc, cargo, ...), so in what way did it help here?